python/shark_turbine/dynamo/importer.py

```diff
@@ -461,53 +461,59 @@ def _import_argument(self, loc: Location, arg: NodeArgument) -> Value:
-            list_operands = []
-            result_type = MlirType.parse(
-                SCALAR_TYPE_TO_TORCH_LIST_TYPE[type(arg[0])], context=self._c
-            ) if type(arg[0]) in SCALAR_TYPE_TO_TORCH_LIST_TYPE else None
-
-            for operand in arg:
-                if not isinstance(operand, type(arg[0])):
-                    raise TypeError(
-                        f"Lists with multiple types are not supported, got: {type(arg[0])}, {type(operand)}"
-                    )
-
-                if isinstance(operand, torch.fx.Node):
-                    if operand in self._multi_result_nodes:
-                        raise RuntimeError(f"Attempt to de-reference a multi-result node")
-                    val = self._v[(operand,0)]
-                    if result_type is None:
-                        list_type: str = str(val.type)
-                        begin_index = 7 if list_type.startswith("!torch.") else None
-                        end_index = list_type.find("<")
-                        end_index = end_index if end_index != -1 else None
-                        list_type = list_type[begin_index:end_index]
-                        result_type = MlirType.parse(f"!torch.list<{list_type}>")
-                else:
-                    val = self._import_default_value(
-                        loc, operand, SCALAR_TYPE_TO_TORCH_TYPE[type(operand)]
-                    )
-
-                list_operands.append(val)
-
-            # construct list op
-            operation = Operation.create(
-                "torch.prim.ListConstruct",
-                results=[result_type],
-                operands=list_operands,
-                loc=loc,
-            )
-            return operation.result
+            return self._import_list_argument(loc, arg)
         elif type(arg) in LITERAL_CONVERTER_MAP._cache:
             with loc:
                 arg_value = LITERAL_CONVERTER_MAP.lookup(type(arg))(arg, self, self._cc)
             return arg_value
         else:
             raise NotImplementedError(f"FIXME: Unsupported Node Argument: {arg}")
 
+    def _import_list_argument(self, loc: Location, arg):
+        # create list operands
+        list_operands = []
+        result_type = MlirType.parse(
+            SCALAR_TYPE_TO_TORCH_LIST_TYPE[type(arg[0])], context=self._c
+        ) if type(arg[0]) in SCALAR_TYPE_TO_TORCH_LIST_TYPE else None
+
+        for operand in arg:
+            if not isinstance(operand, type(arg[0])):
+                raise TypeError(
+                    f"Lists with multiple types are not supported, got: {type(arg[0])}, {type(operand)}"
+                )
+
+            if isinstance(operand, torch.fx.Node):
+                if operand in self._multi_result_nodes:
+                    raise RuntimeError(f"Attempt to de-reference a multi-result node")
+                val = self._v[(operand,0)]
+                if result_type is None:
+                    list_type: str = str(val.type)
```

Comment:
I don't love that we are text parsing the type here but I'll not resist too much for now. However, can you please at least use a regex vs this indexy string manipulation?

Reply:
Yes, I can do that. Ideally, I'd like to use something like `torchMlirTorchListTypeGet` from [here](https://github.com/llvm/torch-mlir/blob/9a1fae97b1e56464a109206acd3b4be7d63ebfc7/lib/CAPI/TorchTypes.cpp#L132) but I'm not sure how I can access that from the python bindings available in turbine at the moment - do you have any idea how I can go about this? I don't see any bindings generated for torch-mlir (i.e. `_torch_ops_gen.py`) in `iree.compile` by which I might access the torch-mlir `ListType`.

Reply:
String indexing is now refactored as regex

Reply:
I was hoping we wouldn't need most of those because they are complicated and slow with all of the goo (which will make it harder to build a faster interface later).

This importer isn't productivity focused, so I'm fine if it has some hacks that we can clean up later.